pkg/dns/treecache.go

```diff
@@ -199,10 +143,6 @@ func (cache *TreeCache) deletePath(path ...string) bool {
 }
 
 func (cache *TreeCache) deleteEntry(key string, path ...string) bool {
-	if cache.rwLock != nil {
```

Comment:
I don't think the comment would make sense here. the treecache is not thread-safe.
It could be used by the same thread/go-routine without locking,
or by multiple threads, using a simple mutex or rwmutex. 
it is up to the caller to choose..